src/api/Nhs.Appointments.Core/BookingAvailabilityStateService.cs

```diff
@@ -61,22 +131,82 @@ private async Task<BookingAvailabilityState> BuildState(string site, DateTime fr
             }
         }
 
-        //we only need to do this calculation if we want to return the available slots
-        if (returnType == BookingAvailabilityStateReturnType.AvailableSlots)
+        switch (returnType)
         {
-            state.AvailableSlots = slotsList.Where(s => s.Capacity > 0).ToList();
+            case BookingAvailabilityStateReturnType.AvailableSlots:
+                //we only need to do this calculation if we want to return the available slots
+                state.AvailableSlots = slotsList.Where(s => s.Capacity > 0).ToList<SessionInstance>();
+                break;
+            case BookingAvailabilityStateReturnType.WeekSummary:
+                state.WeekSummary = GenerateWeekSummary(bookings, daySummaries);
+                break;
+            case BookingAvailabilityStateReturnType.Recalculations:
+                break;
+            default:
+                throw new ArgumentOutOfRangeException(nameof(returnType), returnType, null);
         }
 
         return state;
     }
 
+    private static WeekSummary GenerateWeekSummary(List<Booking> bookings, List<DaySummary> daySummaries)
+    {
+        foreach (var daySummary in daySummaries)
+        {
+            daySummary.MaximumCapacity = daySummary.Sessions.Sum(x => x.MaximumCapacity);
+            daySummary.RemainingCapacity = daySummary.Sessions.Sum(x => x.RemainingCapacity);
+
+            var bookingsOnDay = bookings.Where(x => DateOnly.FromDateTime(x.From) == daySummary.Date).ToList();
+
+            //includes both supported and orphaned bookings
+            daySummary.BookedAppointments = bookingsOnDay.Count(x => x.Status == AppointmentStatus.Booked);
+
+            //...of which X are orphaned
+            daySummary.OrphanedAppointments = bookingsOnDay.Count(x =>
+                x.Status == AppointmentStatus.Booked &&
+                x.AvailabilityStatus == AvailabilityStatus.Orphaned);
+
+            daySummary.CancelledAppointments = bookingsOnDay.Count(x => x.Status == AppointmentStatus.Cancelled);
```

Comment:
Similarly super pedantic and minor, but unless the compiler rejuggles this we're iterating over `bookingsOnDay` 4 times. Might not read as clearly but doing this in 1 loop should be faster. 

... by like a fraction of a ms, so no real sweat. I think in JS I was doing this was an accumulator, sure dotnet must have an equivalent?